### PR TITLE
Fix copy/rename operations in the perspective editor

### PR DESCRIPTION
**Pull request by dgutierr**
Fixes related to the  copy and rename operations which fail under certain conditions. 

@ederign @tomasdavidorg  mind to take a look?

### CODE REVIEW

File: uberfire-extensions/uberfire-layout-editor/uberfire-layout-editor-backend/src/test/java/org/uberfire/ext/layout/editor/impl/PerspectiveServicesImplTest.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+    @Test
+    public void testCopyToTarget() {
+        Path result = perspectiveServices.copy(path, "newName", path2, "");
+        ArgumentCaptor<LayoutEditorModel> layoutModelArg = ArgumentCaptor.forClass(LayoutEditorModel.class);
+        verify(pluginServices).saveLayout(layoutModelArg.capture(), anyString());
+        LayoutEditorModel layoutModelCopy = layoutModelArg.getValue();
+
+        assertEquals(layoutModelCopy.getName(), "newName");
+        assertEquals(layoutModelCopy.getPath(), result);
```
Comment (jomarko) [inline uberfire-extensions/uberfire-layout-editor/uberfire-layout-editor-backend/src/test/java/org/uberfire/ext/layout/editor/impl/PerspectiveServicesImplTest.java]: I think, here should be also assert that the `result` and `path2` are equal. Or am I missing something?

**Comment by tomasdavidorg** [comment]
Jenkins execute full downstream build

**Comment by tomasdavidorg** [comment]
Hi @dgutierr, I tried this locally and I got this error when I tried to save a page (plugin). Could you please look at it. Thanks.
screenshot from 2017-11-02 07-20-10

File: uberfire-extensions/uberfire-layout-editor/uberfire-layout-editor-backend/src/test/java/org/uberfire/ext/layout/editor/impl/PerspectiveServicesImplTest.java (hunk 1 above)
Comment (dgutierr) [reply uberfire-extensions/uberfire-layout-editor/uberfire-layout-editor-backend/src/test/java/org/uberfire/ext/layout/editor/impl/PerspectiveServicesImplTest.java]: Done.Thanx!

**Comment by dgutierr** [comment]
@tomasdavidorg I'm not able to reproduce that error sorry. Can you give it another try please?

**Comment by tomasdavidorg** [comment]
Jenkins execute full downstream build

**Comment by tomasdavidorg** [comment]
@dgutierr https://issues.jboss.org/browse/AF-901

**Pull request merged**